### PR TITLE
Adding C++ linter to CI

### PR DESCRIPTION
**Pull request by ruck314**
### Description
- Improves the code quality by running the linter automatically for every commit via Github action

### CODE REVIEW

**Comment by ruck314** [comment]
Completed a hardware regression test on 10GbE RUDP example with XVC running.  Seems to work and no change in behavior
image

File: include/rogue/EnableSharedFromThis.h [hunk 1]
```diff
@@ -28,13 +25,13 @@
...
 class EnableSharedFromThisBase : public std::enable_shared_from_this<rogue::EnableSharedFromThisBase> {
-  public:
+ public:
     virtual ~EnableSharedFromThisBase() {}
 };
 
 template <typename T>
 class EnableSharedFromThis : virtual public EnableSharedFromThisBase {
-  public:
+ public:
```
Comment (bengineerd) [inline include/rogue/EnableSharedFromThis.h:34]: Did the linter suggest this spacing?
The previous spacing was set by clang-format.

**Review by bengineerd — COMMENTED** [review]
See code comment.

File: include/rogue/EnableSharedFromThis.h (hunk 1 above)
Comment (ruck314) [reply include/rogue/EnableSharedFromThis.h:34]: https://github.com/slaclab/rogue/commit/cf1635ea5edf421d12c48113246818289e34e362

Comment (ruck314) [reply include/rogue/EnableSharedFromThis.h:34]:
```
resolving c++ linter
public: should be indented +1 space inside class [whitespace/indent]
```

Comment (ruck314) [reply include/rogue/EnableSharedFromThis.h:34]: https://guiquanz.gitbooks.io/google-cc-style-guide/content/ebook/Formatting.html
image

Comment (ruck314) [reply include/rogue/EnableSharedFromThis.h]: [reran clang-format to sync up with cpplint](https://github.com/slaclab/rogue/pull/1012/commits/bbee74378a5052505e45828a7d0823058c54ec7f)

**Review by bengineerd — APPROVED** [review]

**Pull request merged**